Upgrade to cypress v10

DiegoGomez01 opened the pull request:
Hi, Trent Rand. I hope you're well.

These changes are to improve this project to allow the use of cypress v10, Let me know if you have any suggestions. 

## Changes Implemented

- Fix lint errors/warnings
- Upgrade libraries
- Changed the `peerDependencies` to accept cypress >=10 on the package.json
- Changed the flag `integrationFolder` to `specPattern` because that was a change on the new cypress version 10
- Separate the flag `config` and `config-file`. Now to add the configFile, you must include the flag `config-file`, `configFile` or `C` but if you want to add/replace configuration you can do that with `config` or 'c'
- To exclude a file (or list of files) you can use the flag `excludeSpecPattern` and add the files to exclude. i.e `cypress-utils run-parallel --excludeSpecPattern all.cy.js file2.cy.js file3.cy.js`

Thanks for this useful repo. Have a good day.

DiegoGomez01 committed:
Upgrade to cypress v10

trentrand commented on index.js:
```diff
@@ -44,17 +44,28 @@ var argv = yargs.scriptName('cypress-utils')
     type: 'string',
-    description: 'Path to folder containing integration test files',
-    default: 'cypress/integration',
+    description: 'Path to folder containing e2e test files',
```
```suggestion
    description: 'A glob pattern of the test files to load.',
```

trentrand replied:
Pulled this description from https://docs.cypress.io/guides/references/configuration#e2e but reworded it to remove support for Array type.

DiegoGomez01 replied:
Thanks for the suggestion Trent Rand. I'll do the change. It's better.

trentrand commented on index.js:
```diff
@@ -123,7 +134,7 @@ function handleResults(error, results) {
 
   // Clear command-line before printing results
-  process.stdout.write('\033c')
+  process.stdout.write('\x1Bc');
```
I'm hesitant to make this change. I wasn't very familiar with either, but my search led to a warning that `\x1Bc` may not work properly in all situations.

https://stackoverflow.com/a/47503782/2020476

Were you seeing an issue with `\033c`?

DiegoGomez01 replied:
It works fine. But I had an eslint warning. But you're right, maybe it does not work correctly in all situations. I'll revert the change

trentrand commented on index.js:
```diff
@@ -44,17 +44,28 @@ var argv = yargs.scriptName('cypress-utils')
+    global: true,
+  })
+  .option('excludeSpecPattern', {
+    type: 'array',
```
```suggestion
    type: 'array',
    default: [],
```

DiegoGomez01 replied:
Yes, that change it's important. Many thanks

trentrand commented on index.js:
```diff
@@ -159,59 +170,37 @@ function printResultsForCommand(subjectName, subjectResults, command) {
       );
-    })
+    });
+  if (argv.excludeSpecPattern.length > 0) {
```
this could fail if `argv.excludeSpecPattern` is undefined

DiegoGomez01 replied:
I'm not sure because we're including a default value `[]` in the excludeSpecPattern. May I include the validation anyway?

trentrand replied:
You're right, the code change in [58d39db](https://github.com/trentrand/cypress-utils/pull/25/commits/58d39db353ae1decd42a55df87927648b18d2b67) fixes this

trentrand commented on package.json:
```diff
@@ -25,16 +25,16 @@
   },
   "devDependencies": {
-    "cypress": "8.3.0"
+    "cypress": "10.7.0"
```
maybe `^10.0.0` to support a wider range of users?

DiegoGomez01 replied:
Yes, sure.

trentrand reviewed (commented):
Thank you very much for the contribution!

It looks great, just a few minor points we can address before publishing!

DiegoGomez01 committed:
Change cypress devDependencies to ^10

DiegoGomez01 committed:
Change description on specPattern

DiegoGomez01 committed:
Add default value [] on excludeSpecPattern

DiegoGomez01 committed:
Revert change on process.stdout.write

trentrand reviewed (approved):
Well done! 👏🏻 I appreciate you contributing these changes.

I'll merge and publish version 3.0.0 of the package.

trentrand commented:
I tested this against the [cypress-kitchen-sink](https://github.com/cypress-io/cypress-example-kitchensink) test files.

`cypress-utils run-parallel 1-getting-started/todo 2-advanced-examples/waiting viewport window`
> ![image](https://user-images.githubusercontent.com/6083067/190932112-b01f4807-48f3-42b3-8e5d-4806cda9bc95.png)

trentrand committed:
Add docs and code examples for new Cypress v10 CLI options

trentrand committed:
Use minimatch when handling `excludeSpecPattern` command-line option

These changes update the filtering implementation to use 'minimatch',
which matches how Cypress filters with this command-line option
internally.

This guarantees that the same values will work in Cypress Module API and
cypress-utils.

trentrand commented:
@DiegoGomez01, I amended commits 60f2f11 and ad50f08

trentrand commented:
These changes were published as version 3.0.0! 🎉 

See the release for more details:
https://github.com/trentrand/cypress-utils/releases/tag/v3.0.0
